src/prerender.ts

```diff
@@ -245,6 +248,13 @@ export async function prerender(nitro: Nitro) {
     interval: nitro.options.prerender.interval,
   });
 
+  await nitro.hooks.callHook("prerenderer:done", {
+    nitro,
+    prerenderer: nitroRenderer,
+    erroredRoutes,
+    generateRoute,
```

Comment:
Allowing `generateRoute` here is kinda tricky. Because we are declaring prerendering is "done" to "all" listeners and with final errors.

We might expose API in other different ways. One way would be to pass it to second arg of `prerender:init(prerenderer, { generateRoute }`.

I understand this is final goal of your PR and init hook might not be best fitting as well. Because of this i would suggest to discuss and introduce this in next PR (we could think of a prerender controller object being provided earlier)

Reply:
Make sense if we're renaming it and exposing a prerender controller seems like a good DX

Reply:
To clarify, I don't think it's worth adding generateRoute on the init hook, better to just wait until the prerenderer controller is available

Reply:
Let's do this then! (feel free to open a new discussion issue to not forget)